Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyPostedInvoiceExport.Codeunit.al

```diff
@@ -0,0 +1,459 @@
+    begin
+        GeneralLedgerSetup.Get();
+
+        if ShpfyShop."Currency Code" = '' then
+            ShopifyLocalCurrencyCode := GeneralLedgerSetup."LCY Code"
+        else
+            ShopifyLocalCurrencyCode := ShpfyShop."Currency Code";
+
+        if SalesInvoiceHeader."Currency Code" = '' then
+            exit(ShopifyLocalCurrencyCode = GeneralLedgerSetup."LCY Code")
+        else
+            exit(ShopifyLocalCurrencyCode = SalesInvoiceHeader."Currency Code");
+    end;
+
+    local procedure ShopifyPaymentTermsExists(PaymentTermsCode: Code[10]): Boolean
+    var
+        ShpfyPaymentTerms: Record "Shpfy Payment Terms";
+    begin
+        ShpfyPaymentTerms.SetRange("Payment Terms Code", PaymentTermsCode);
+        ShpfyPaymentTerms.SetRange("Shop Code", ShpfyShop.Code);
+
+        if not ShpfyPaymentTerms.FindFirst() then begin
+            ShpfyPaymentTerms.SetRange("Payment Terms Code");
+            ShpfyPaymentTerms.SetRange("Is Primary", true);
+
+            if not ShpfyPaymentTerms.FindFirst() then
+                exit(false);
+        end;
+
+        exit(true);
+    end;
+
+    local procedure CheckCustomerTemplates(CustomerNo: Code[20]): Boolean
+    var
+        ShpfyCustomerTemplate: Record "Shpfy Customer Template";
+    begin
+        ShpfyCustomerTemplate.SetRange("Default Customer No.", CustomerNo);
+        ShpfyCustomerTemplate.SetRange("Shop Code", ShpfyShop.Code);
+        exit(not ShpfyCustomerTemplate.IsEmpty());
+    end;
+
+    local procedure CheckSalesInvoiceHeaderLines(SalesInvoiceHeader: Record "Sales Invoice Header"): Boolean
+    var
+        SalesInvoiceLine: Record "Sales Invoice Line";
+    begin
+        SalesInvoiceLine.SetFilter(Type, '<>%1', SalesInvoiceLine.Type::" ");
+        if SalesInvoiceLine.IsEmpty() then
+            exit(false);
+
+        SalesInvoiceLine.Reset();
+
+        SalesInvoiceLine.SetRange("Document No.", SalesInvoiceHeader."No.");
+        SalesInvoiceLine.SetRange(Type, SalesInvoiceLine.Type::Item);
+        if SalesInvoiceLine.FindSet() then
+            repeat
+                if (SalesInvoiceLine.Quantity <> 0) and (SalesInvoiceLine.Quantity <> Round(SalesInvoiceLine.Quantity, 1)) then
+                    exit(false);
+
+                if ShpfyShop."Items Mapped to Products" then
+                    if not ItemIsMappedToShopifyProduct(SalesInvoiceLine) then
+                        exit(false);
+
+                if (SalesInvoiceLine.Type <> SalesInvoiceLine.Type::" ") and (SalesInvoiceLine."No." = '') then
+                    exit(false);
+            until SalesInvoiceLine.Next() = 0;
+
+        exit(true);
+    end;
+
+    local procedure SetSalesInvoiceShopifyOrderInformation(var SalesInvoiceHeader: Record "Sales Invoice Header"; OrderId: BigInteger; OrderNo: Code[50])
+    begin
+        SalesInvoiceHeader.Validate("Shpfy Order Id", OrderId);
+        SalesInvoiceHeader.Validate("Shpfy Order No.", OrderNo);
+        SalesInvoiceHeader.Modify(true);
```

Comment:
If we modify the SalesHeader then the codedunit should have the permission that indirect modify permission is sufficient.